### PR TITLE
AHC-1066 Add situations to categories in admin ui

### PR DESCRIPTION
**Pull request by jennywagg**

### CODE REVIEW

File: app/helpers/admin/form_helper.rb [hunk 1]
```diff
@@ -49,14 +49,25 @@ def class_name_for(category)
...
 
+    # rubocop:disable Metrics/MethodLength
     def checkbox_tag_for(category)
-      check_box_tag(
-        'service[category_ids][]',
-        category.id,
-        @taxonomy_ids.include?(category.taxonomy_id),
-        id: "category_#{category.taxonomy_id}"
-      )
+      if category.type == 'service'
```
Comment (solipet) [inline app/helpers/admin/form_helper.rb]: Could tighten this up a bit, e.g.,
```
cb_name = category.type == 'service' ? 'service[category_ids][]' : 'situation[category_ids][]'
check_box_tag(
  cb_name,
  category.id,
  @taxonomy_ids.include?(category.taxonomy_id),
  id: "category_#{category.taxonomy_id}"
)
```

File: app/views/admin/services/_confirm_delete_service.html.haml [hunk 2]
```diff
@@ -15,4 +15,4 @@
           #{service_name || service_id}.
     .modal-footer
       %button.btn.btn-default{ 'data-dismiss' => 'modal' }  Close
-      = link_to t('admin.buttons.confirm_delete_service'), { action: :destroy, id: service_id, name: service_name }, method: :delete, class: 'btn btn-danger'
+      = link_to t('admin.buttons.confirm_delete_service'), admin_location_service_path(location_id: location_id, id: service_id), { action: :destroy, class: 'btn btn-danger',  method: :delete }
```
Comment (solipet) [inline app/views/admin/services/_confirm_delete_service.html.haml]: Do we need both `action: :destroy` *and* `method: :delete`?

File: app/views/admin/services/forms/_categories.html.haml [hunk 3]
```diff
@@ -7,4 +7,9 @@
 
   = hidden_field_tag 'service[category_ids][]', nil
   = field_set_tag nil, id: 'categories' do
-    = nested_categories(Category.arrange(order: :taxonomy_id))
+    = f.label :services
+    = nested_categories(Category.where(type: 'service').arrange(order: :taxonomy_id))
```
Comment (solipet) [inline app/views/admin/services/forms/_categories.html.haml]: could add scopes to Category model, e.g.:
```
scope :service, -> { where(type: 'service') }
```
Then you can do `Category.services.arrange...`

File: app/views/admin/services/_confirm_delete_service.html.haml (hunk 2 above)
Comment (jennywagg) [reply app/views/admin/services/_confirm_delete_service.html.haml]: yes 😄not sure why!

**Pull request merged**